packages/scandipwa/src/component/ProductBundleItems/ProductBundleItems.container.js

```diff
@@ -139,6 +153,55 @@ export class ProductBundleItemsContainer extends ProductCustomizableOptionsConta
             );
     }
 
+    resetDefaultSelectOptions() {
+        this.setState({
+            selectedCheckboxValues: [],
+            selectedDropdownOptions: []
+        }, () => this.setDefaultSelectedOptions());
+    }
+
+    setDefaultSelectedOptions() {
+        const { items } = this.props;
+
+        return items.reduce((acc, item) => {
+            const { type } = item;
+
+            switch (type) {
+            case SELECT_TYPE:
+            case RADIO_TYPE: // handle radio as select
+                this.setDefaultValue(item, SELECT_TYPE);
+                break;
+            case CHECKBOX_TYPE:
+            case MULTI_TYPE: // handle multi-select as checkbox
+                this.setDefaultValue(item);
+                break;
+            default:
+                return acc;
+            }
+
+            return acc;
+        }, []);
+    }
+
+    setDefaultValue(item, type) {
+        const { option_id, options } = item;
+
+        return options.reduce((acc, { is_default, id, quantity }) => {
+            if (is_default) {
+                const value = id.toString();
+
+                if (type === SELECT_TYPE) {
+                    this.setSelectedDropdownValue(option_id, { value, quantity });
+                    return acc;
+                }
+
+                this.setSelectedCheckboxValues(option_id, { value, quantity });
+            }
+
+            return acc;
```

Comment:
What's the idea here? It seems to me that the logic with and without if statement is the same - branching is not necessary.

Reply:
@yeegor 
If type is 'select' then we do function setSelectedDropdownValue, if no then setSelectedCheckboxValues. I guess we are talking about this ?